Add Xyce 7.4 (second attempt)

fedeinthemix opened the pull request:
###### Motivation for this change
Added the analog circuit simulator Xyce previously not present in `nixpkgs`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

fedeinthemix committed:
maintainers: add fbeffa

fedeinthemix committed:
trilinos: init at version 12.12.1

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>
                Kim Lindberger <kim.lindberger@gmail.com>

fedeinthemix committed:
xyce,xyce-parallel: init at 7.4

Co-authored-by: Sandro <sandro.jaeckel@gmail.com> and
                Kim Lindberger <kim.lindberger@gmail.com>

fedeinthemix commented:
@bgamari @rongcuid I see that you are the maintainers of `ngspice`. Maybe you are interested in reviewing this addition of `Xyce` :-)

fedeinthemix commented:
I believe that failing of "trilinos, trilinos.passthru.tests, xyce, xyce.passthru.tests on x86_64-linux" was due to an unrelated commit in master that in the mean time has been fixed. I've re-tested my patch on master commit f9a214044be750fb8e186b21e4dbe90ad2e2c0b2 and on my machine (NixOS 21.11 linux-x86_64 )it built successfully.

fedeinthemix commented:
Hi @talyz, my PR is ready for review. Is somebody going to check it, or should I notify somebody?

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/712

SuperSandro2000 commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,159 @@
+, enableTests ? true
+}:
+
+stdenv.mkDerivation rec {
+
+  pname = "xyce";
```
```suggestion
stdenv.mkDerivation rec {
  pname = "xyce";
```

SuperSandro2000 commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,159 @@
+
+  pname = "xyce";
+  version = "7.4.0";
+  twoLevelVersion = with builtins;
+    concatStringsSep  "." (lib.lists.take 2 (splitVersion version));
```
```suggestion
```

SuperSandro2000 commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,159 @@
+  postInstall = if enableDocs then ''
+    local docFiles=("doc/Users_Guide/Xyce_UG"
+      "doc/Reference_Guide/Xyce_RG"
+      "doc/Release_Notes/Release_Notes_${twoLevelVersion}/Release_Notes_${twoLevelVersion}")
```
```suggestion
      "doc/Release_Notes/Release_Notes_${lib.versions.majrMinor version}/Release_Notes_${lib.versions.majrMinor version}")
```

SuperSandro2000 commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,159 @@
+      rev = "82f96bbe05bac5921cd6fa1e8bb6a2983a797bf8";
+      sha256 = "sha256-sOHjQEo4FqlDseTtxFVdLa0SI/VAf2OkwQV7QSL7SNM=";
+    })
+    (fetchurl {
```
This shouldn't be the case here like above. Please try using fetchFromGitHub.

SuperSandro2000 commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,159 @@
+    })
+  ];
+
+  sourceRoot = with builtins; "./Xyce-${substring 0 7 (head srcs).rev}";
```
```suggestion
  sourceRoot = "./Xyce-${lib.substring 0 7 (lib.head srcs).rev}";
```

SuperSandro2000 commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,159 @@
+
+  preConfigure = "./bootstrap";
+
+  configureFlags = [ "CXXFLAGS=-O3"
+                     "--enable-xyce-shareable"
+                     "--enable-shared"
+                     "--enable-stokhos"
+                     "--enable-amesos2"
+                   ] ++ lib.optionals withMPI [
+                     "--enable-mpi"
+                     "CXX=mpicxx"
+                     "CC=mpicc"
+                     "F77=mpif77"
+                   ];
```
```suggestion
  configureFlags = [
    "CXXFLAGS=-O3"
    "--enable-xyce-shareable"
    "--enable-shared"
    "--enable-stokhos"
    "--enable-amesos2"
  ] ++ lib.optionals withMPI [
    "--enable-mpi"
    "CXX=mpicxx"
    "CC=mpicc"
    "F77=mpif77"
  ];
```

SuperSandro2000 commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,159 @@
+                     "F77=mpif77"
+                   ];
+
+  nativeBuildInputs = [ autoconf automake gfortran libtool_2
+                      ] ++ lib.optionals enableDocs [
+                        (texlive.combine {
+                          inherit (texlive)
+                            scheme-medium
+                            koma-script
+                            optional
+                            framed
+                            enumitem
+                            multirow
+                            preprint; })
+                      ];
```
```suggestion
  nativeBuildInputs = [
    autoconf automake gfortran libtool_2
  ] ++ lib.optionals enableDocs [
    (texlive.combine {
      inherit (texlive)
        scheme-medium
        koma-script
        optional
        framed
        enumitem
        multirow
        preprint;
      })
  ];
```
Is there no already build texlive package that is close enough to what we need? If so please use that.

fedeinthemix replied:
I've tried several packages, but have not found one working.

SuperSandro2000 commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,159 @@
+                            preprint; })
+                      ];
+
+  buildInputs = [ bison blas flex fftw lapack suitesparse trilinos
+                ] ++ lib.optionals withMPI [ mpi ];
```
```suggestion
  buildInputs = [
    bison blas flex fftw lapack suitesparse trilinos
  ] ++ lib.optionals withMPI [ mpi ];
```

SuperSandro2000 commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,159 @@
+  postPatch = ''
+    pushd ../Xyce_Regression-Release-${version}
+    find Netlists -type f -regex ".*\.sh\|.*\.pl" -exec chmod ugo+x {} \;
+    patchShebangs Netlists/
+    patchShebangs TestScripts/
```
```suggestion
    patchShebangs Netlists/ TestScripts/
```

SuperSandro2000 commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,159 @@
+    patchShebangs Netlists/
+    patchShebangs TestScripts/
+    # patch script generating functions
+    sed -i -E 's/\/usr\/bin\/env perl/${builtins.replaceStrings [ "." "/" "-" "!" ] [ "\\." "\\/" "\\-" "\\!" ] perl.outPath}\/bin\/perl/' TestScripts/XyceRegression/Testing/Netlists/RunOptions/runOptions.cir.sh
+    sed -i -E 's/\/bin\/sh/${builtins.replaceStrings [ "." "/" "-" "!" ] [ "\\." "\\/" "\\-" "\\!" ] bash.outPath}\/bin\/sh/' TestScripts/XyceRegression/Testing/Netlists/RunOptions/runOptions.cir.sh
```
```suggestion
    sed -i -E 's/\/usr\/bin\/env perl/${builtins.replaceStrings [ "." "/" "-" "!" ] [ "\\." "\\/" "\\-" "\\!" ] perl.outPath}\/bin\/perl/'  \
      TestScripts/XyceRegression/Testing/Netlists/RunOptions/runOptions.cir.sh
    sed -i -E 's/\/bin\/sh/${builtins.replaceStrings [ "." "/" "-" "!" ] [ "\\." "\\/" "\\-" "\\!" ] bash.outPath}\/bin\/sh/' \
      TestScripts/XyceRegression/Testing/Netlists/RunOptions/runOptions.cir.sh
```

SuperSandro2000 commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,159 @@
+    popd
+  '';
+
+  checkInputs = [ bc
+                  perl
+                  perlPackages.DigestMD5
+                  (python38.withPackages (ps: with ps; [ numpy scipy ]))
+                ] ++ lib.optionals withMPI [ mpi openssh ];
```
```suggestion
  checkInputs = [
    bc
    perl
    perlPackages.DigestMD5
    (python38.withPackages (ps: with ps; [ numpy scipy ]))
  ] ++ lib.optionals withMPI [ mpi openssh ];
```

Why can't we use python3 which points to python39? python38 is no longer good maintained might just fall apart next week.

fedeinthemix replied:
That's an oversight of mine. Will change that.

SuperSandro2000 commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,159 @@
+
+  checkPhase = ''
+    XYCE_BINARY="$(pwd)/src/Xyce"
+    EXECSTRING="${if withMPI then "mpirun -np 2 " else ""}$XYCE_BINARY"
```
```suggestion
    EXECSTRING="${lib.optionalString withMPI "mpirun -np 2 "}$XYCE_BINARY"
```

SuperSandro2000 commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,159 @@
+      --xyce_test="''${TEST_ROOT}" \
+      --taglist="$($TEST_ROOT/TestScripts/suggestXyceTagList.sh "$XYCE_BINARY" | sed -E -e 's/TAGLIST=([^ ]+).*/\1/' -e '2,$d')" \
+      --resultfile="$(pwd)/test_results" \
+      ${if withMPI then "--excludelist=$EXLUDE_TESTS_FILE" else ""} \
```
```suggestion
      ${lib.optionalString withMPI "--excludelist=$EXLUDE_TESTS_FILE"} \
```

SuperSandro2000 commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,159 @@
+
+  outputs = [ "out" "doc" ];
+
+  postInstall = if enableDocs then ''
```
```suggestion
  postInstall = lib.optionalString enableDocs ''
```

SuperSandro2000 commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,159 @@
+      install -t $doc/share/doc/${pname}-${version}/ $(basename $d.pdf)
+      popd
+    done
+    '' else "";
```
```suggestion
    '';
```

SuperSandro2000 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -32252,6 +32256,11 @@ with pkgs;
 
+  xyce = callPackage ../applications/science/electronics/xyce { };
+
+  xyce-parallel = callPackage ../applications/science/electronics/xyce
+    { withMPI = true; trilinos = trilinos-mpi; };
```
```suggestion
  xyce-parallel = callPackage ../applications/science/electronics/xyce {
    withMPI = true;
    trilinos = trilinos-mpi;
  };
```

SuperSandro2000 commented on pkgs/development/libraries/science/math/trilinos/default.nix:
```diff
@@ -0,0 +1,94 @@
+
+  buildInputs = [ blas boost lapack suitesparse ] ++ lib.optionals withMPI [ mpi ];
+
+  preConfigure = if withMPI then
+                   ''cmakeFlagsArray+=(${flagsBase} ${flagsParallel})''
+                 else
+                   ''cmakeFlagsArray+=(${flagsBase})'';
```
```suggestion
  preConfigure = if withMPI then
    ''cmakeFlagsArray+=(${flagsBase} ${flagsParallel})''
  else
    ''cmakeFlagsArray+=(${flagsBase})'';
```
Why is this not using cmakeFlags?

SuperSandro2000 commented on pkgs/development/libraries/science/math/trilinos/default.nix:
```diff
@@ -0,0 +1,94 @@
+  version = "12.12.1"; # Xyce 7.4 requires version 12.12.1
+
+  src = fetchurl {
+    url = "https://github.com/trilinos/Trilinos/archive/refs/tags/${pname}-release-"
+          + builtins.replaceStrings [ "." ] [ "-" ] version
+          + ".tar.gz";
```
```suggestion
    url = "https://github.com/trilinos/Trilinos/archive/refs/tags/${pname}-release-${lib.replaceStrings [ "." ] [ "-" ] version}.tar.gz";
```

SuperSandro2000 commented on pkgs/development/libraries/science/math/trilinos/default.nix:
```diff
@@ -0,0 +1,94 @@
+        -DTPL_ENABLE_MPI=ON
+      '';
+
+in stdenv.mkDerivation rec {
+
+  pname = "trilinos";
```
```suggestion
in stdenv.mkDerivation rec {
  pname = "trilinos";
```

SuperSandro2000 commented on pkgs/development/libraries/science/math/trilinos/default.nix:
```diff
@@ -0,0 +1,94 @@
+}:
+
+# NOTE: Not all packages are enabled.  We specifically enable the ones
+# required to build Xyce.  If the need comes, we can enable more of them.
```
```suggestion
# required to build Xyce. If the need comes, we can enable more of them.
```

SuperSandro2000 commented on pkgs/development/libraries/science/math/trilinos/default.nix:
```diff
@@ -0,0 +1,94 @@
+let flagsBase = ''
+        -G "Unix Makefiles"
+        -DBUILD_SHARED_LIBS=ON
+        -DCMAKE_CXX_FLAGS="-O3 -fPIC"
+        -DCMAKE_C_FLAGS="-O3 -fPIC"
+        -DCMAKE_Fortran_FLAGS="-O3 -fPIC"
+        -DTrilinos_ENABLE_NOX=ON
+        -DNOX_ENABLE_LOCA=ON
+        -DTrilinos_ENABLE_EpetraExt=ON
+        -DEpetraExt_BUILD_BTF=ON
+        -DEpetraExt_BUILD_EXPERIMENTAL=ON
+        -DEpetraExt_BUILD_GRAPH_REORDERINGS=ON
+        -DTrilinos_ENABLE_TrilinosCouplings=ON
+        -DTrilinos_ENABLE_Ifpack=ON
+        -DTrilinos_ENABLE_AztecOO=ON
+        -DTrilinos_ENABLE_Belos=ON
+        -DTrilinos_ENABLE_Teuchos=ON
+        -DTeuchos_ENABLE_COMPLEX=ON
+        -DTrilinos_ENABLE_Amesos=ON
+        -DAmesos_ENABLE_KLU=ON
+        -DTrilinos_ENABLE_Amesos2=ON
+        -DAmesos2_ENABLE_KLU2=ON
+        -DAmesos2_ENABLE_Basker=ON
+        -DTrilinos_ENABLE_Sacado=ON
+        -DTrilinos_ENABLE_Stokhos=ON
+        -DTrilinos_ENABLE_Kokkos=ON
+        -DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES=OFF
+        -DTrilinos_ENABLE_CXX11=ON
+        -DTPL_ENABLE_AMD=ON
+        -DTPL_ENABLE_BLAS=ON
+        -DTPL_ENABLE_LAPACK=ON
+    '';
+    flagsParallel = ''
+        -DCMAKE_C_COMPILER=mpicc
+        -DCMAKE_CXX_COMPILER=mpic++
+        -DCMAKE_Fortran_COMPILER=mpif77
+        -DTrilinos_ENABLE_Isorropia=ON
+        -DTrilinos_ENABLE_Zoltan=ON
+        -DTPL_ENABLE_MPI=ON
+      '';
```
```suggestion
let
  flagsBase = ''
    -G "Unix Makefiles"
    -DBUILD_SHARED_LIBS=ON
    -DCMAKE_CXX_FLAGS="-O3 -fPIC"
    -DCMAKE_C_FLAGS="-O3 -fPIC"
    -DCMAKE_Fortran_FLAGS="-O3 -fPIC"
    -DTrilinos_ENABLE_NOX=ON
    -DNOX_ENABLE_LOCA=ON
    -DTrilinos_ENABLE_EpetraExt=ON
    -DEpetraExt_BUILD_BTF=ON
    -DEpetraExt_BUILD_EXPERIMENTAL=ON
    -DEpetraExt_BUILD_GRAPH_REORDERINGS=ON
    -DTrilinos_ENABLE_TrilinosCouplings=ON
    -DTrilinos_ENABLE_Ifpack=ON
    -DTrilinos_ENABLE_AztecOO=ON
    -DTrilinos_ENABLE_Belos=ON
    -DTrilinos_ENABLE_Teuchos=ON
    -DTeuchos_ENABLE_COMPLEX=ON
    -DTrilinos_ENABLE_Amesos=ON
    -DAmesos_ENABLE_KLU=ON
    -DTrilinos_ENABLE_Amesos2=ON
    -DAmesos2_ENABLE_KLU2=ON
    -DAmesos2_ENABLE_Basker=ON
    -DTrilinos_ENABLE_Sacado=ON
    -DTrilinos_ENABLE_Stokhos=ON
    -DTrilinos_ENABLE_Kokkos=ON
    -DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES=OFF
    -DTrilinos_ENABLE_CXX11=ON
    -DTPL_ENABLE_AMD=ON
    -DTPL_ENABLE_BLAS=ON
    -DTPL_ENABLE_LAPACK=ON
  '';
  flagsParallel = ''
    -DCMAKE_C_COMPILER=mpicc
    -DCMAKE_CXX_COMPILER=mpic++
    -DCMAKE_Fortran_COMPILER=mpif77
    -DTrilinos_ENABLE_Isorropia=ON
    -DTrilinos_ENABLE_Zoltan=ON
    -DTPL_ENABLE_MPI=ON
  '';
```

SuperSandro2000 reviewed (commented):
Mostly formatting and style things to follow current stylistic recommendations.

SuperSandro2000 commented:
Please don't forget to squash the commit history together that it looks a lot nicer.

fedeinthemix commented:
Hi @SuperSandro2000, thanks for the review. I merged your changes and squashed the commits.

Regarding your note about `fetchFromGitHub`: I confirm that using `fetchFromGitHub` doesn't retrieve the documentation due to the file https://github.com/Xyce/Xyce/blob/master/doc/.gitattributes, see also https://groups.google.com/g/xyce-users/c/MxeViRo8CT4. However, it works for the regression tests and so I changed that.
I was surprised in doing this change to find out that the permission of files are different and this caused some failures. To fix those I slightly modified the `postPatch` phase to restore necessary permissions.

SuperSandro2000 commented:
I am currently pushing little fixes to your commit messages. After that I think we can merge,

SuperSandro2000 commented:
xyce fails to build

```
Testing restricted mode for {,r}epstopdf
!!! Error: Option forbidden in restricted mode: --gscmd
resolvers       | resolving | variable 'SELFAUTOLOC' set to '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/bin'
resolvers       | resolving | variable 'SELFAUTODIR' set to '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021'
resolvers       | resolving | variable 'SELFAUTOPARENT' set to '/nix/store'
resolvers       | resolving | variable 'TEXMFCNF' set to '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf/web2c'
resolvers       | resolving | variable 'TEXMF' set to ''
resolvers       | resolving | variable 'TEXOS' set to 'n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021'
resolvers       | resolving |
resolvers       | resolving | using configuration specification '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf/web2c'
resolvers       | resolving |
resolvers       | resolving | looking for regular 'texmfcnf.lua' on given path '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf/web2c' from specification '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf/web2c'
resolvers       | resolving | found regular configuration file '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf/web2c/texmfcnf.lua'
resolvers       | resolving |
resolvers       | resolving | loading configuration file '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf/web2c/texmfcnf.lua'
resolvers       | resolving |
resolvers       | resolving | locating list of 'home:.texlive2021/texmf-config' (runtime) (tree:///home:.texlive2021/texmf-config)
resolvers       | methods | resolving, method 'locators', how 'uri', handler 'tree', argument 'tree:///home:.texlive2021/texmf-config'
resolvers       | trees | locator 'home:.texlive2021/texmf-config' not found
resolvers       | resolving | locating list of 'home:texmf' (runtime) (tree:///home:texmf)
resolvers       | methods | resolving, method 'locators', how 'uri', handler 'tree', argument 'tree:///home:texmf'
resolvers       | trees | locator 'home:texmf' not found
resolvers       | resolving | locating list of '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-config' (cached)
resolvers       | methods | resolving, method 'locators', how 'uri', handler 'file', argument '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-config'
resolvers       | files | file locator '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-config' found as '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-config'
resolvers       | resolving | hash '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-config' appended
resolvers       | resolving | locating list of '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var' (cached)
resolvers       | methods | resolving, method 'locators', how 'uri', handler 'file', argument '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var'
resolvers       | files | file locator '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var' found as '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var'
resolvers       | resolving | hash '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var' appended
resolvers       | resolving | locating list of '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-project' (cached)
resolvers       | methods | resolving, method 'locators', how 'uri', handler 'file', argument '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-project'
resolvers       | files | file locator '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-project' not found
resolvers       | resolving | locating list of '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-fonts' (cached)
resolvers       | methods | resolving, method 'locators', how 'uri', handler 'file', argument '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-fonts'
resolvers       | files | file locator '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-fonts' not found
resolvers       | resolving | locating list of '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-local' (cached)
resolvers       | methods | resolving, method 'locators', how 'uri', handler 'file', argument '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-local'
resolvers       | files | file locator '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-local' found as '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-local'
resolvers       | resolving | hash '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-local' appended
resolvers       | resolving | locating list of '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf' (cached)
resolvers       | methods | resolving, method 'locators', how 'uri', handler 'file', argument '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf'
resolvers       | files | file locator '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf' found as '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf'
resolvers       | resolving | hash '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf' appended
resolvers       | resolving |
resolvers       | methods | resolving, method 'generators', how 'uri', handler 'file', argument '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-config'
resolvers       | expansions | scanning path '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-config', branch '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-config'
resolvers       | expansions | 2 files found on 1 directories with 1 uppercase remappings
resolvers       | methods | resolving, method 'generators', how 'uri', handler 'file', argument '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var'
resolvers       | expansions | scanning path '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var', branch '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var'
resolvers       | expansions | 76 files found on 17 directories with 1 uppercase remappings
resolvers       | methods | resolving, method 'generators', how 'uri', handler 'file', argument '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-local'
resolvers       | expansions | scanning path '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-local', branch '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-local'
resolvers       | expansions | 1 files found on 0 directories with 1 uppercase remappings
resolvers       | methods | resolving, method 'generators', how 'uri', handler 'file', argument '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf'
resolvers       | expansions | scanning path '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf', branch '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf'
resolvers       | globbing | confusing filename, name: 'custMatTransform.py', lower: 'custmattransform.py', already: 'CustMatTransform.py'
resolvers       | globbing | confusing filename, name: 'setCustomAnchor.py', lower: 'setcustomanchor.py', already: 'SetCustomAnchor.py'
resolvers       | globbing | confusing filename, name: 'widg_addLabel.py', lower: 'widg_addlabel.py', already: 'Widg_addLabel.py'
resolvers       | globbing | confusing filename, name: 'widg_addPolyOpt.py', lower: 'widg_addpolyopt.py', already: 'Widg_addPolyOpt.py'
resolvers       | globbing | confusing filename, name: 'widg_editBezier.py', lower: 'widg_editbezier.py', already: 'Widg_editBezier.py'
resolvers       | expansions | 27938 files found on 2438 directories with 3621 uppercase remappings
resolvers       | resolving |
resolvers       | resolving | saving tree '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-config'
resolvers       | caches | hashing tree '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-config', hash '3b2575f069016ddeaabc9383b17339f2'
resolvers       | caches | hashing tree '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf/web2c/texmfcnf.lua', hash '1dbe71212a391f836718d0dfc155f8d1'
resolvers       | caching | preparing 'files' for '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-config'
resolvers       | caching | category 'files', cachename '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-config' saved in '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/3b2575f069016ddeaabc9383b17339f2.lua'
system          | lua | compiling '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/3b2575f069016ddeaabc9383b17339f2.lua' into '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/3b2575f069016ddeaabc9383b17339f2.luc'
system          | lua | dumping '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/3b2575f069016ddeaabc9383b17339f2.lua' into '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/3b2575f069016ddeaabc9383b17339f2.luc' stripped
resolvers       | caching | 'files' compiled to '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/3b2575f069016ddeaabc9383b17339f2.luc'
resolvers       | resolving | saving tree '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var'
resolvers       | caches | hashing tree '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var', hash 'fa3932c55184ae5e300dece358aaf0e4'
resolvers       | caching | preparing 'files' for '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var'
resolvers       | caching | category 'files', cachename '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var' saved in '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/fa3932c55184ae5e300dece358aaf0e4.lua'
system          | lua | compiling '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/fa3932c55184ae5e300dece358aaf0e4.lua' into '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/fa3932c55184ae5e300dece358aaf0e4.luc'
system          | lua | dumping '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/fa3932c55184ae5e300dece358aaf0e4.lua' into '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/fa3932c55184ae5e300dece358aaf0e4.luc' stripped
resolvers       | caching | 'files' compiled to '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/fa3932c55184ae5e300dece358aaf0e4.luc'
resolvers       | resolving | saving tree '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-local'
resolvers       | caches | hashing tree '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-local', hash '1930a27f3a8ec5f390f25bb2e7b72c7f'
resolvers       | caching | preparing 'files' for '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-local'
resolvers       | caching | category 'files', cachename '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-local' saved in '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/1930a27f3a8ec5f390f25bb2e7b72c7f.lua'
system          | lua | compiling '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/1930a27f3a8ec5f390f25bb2e7b72c7f.lua' into '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/1930a27f3a8ec5f390f25bb2e7b72c7f.luc'
system          | lua | dumping '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/1930a27f3a8ec5f390f25bb2e7b72c7f.lua' into '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/1930a27f3a8ec5f390f25bb2e7b72c7f.luc' stripped
resolvers       | caching | 'files' compiled to '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/1930a27f3a8ec5f390f25bb2e7b72c7f.luc'
resolvers       | resolving | saving tree '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf'
resolvers       | caches | hashing tree '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf', hash '28a03cdd14c63e5c76566ae30b8a381b'
resolvers       | caching | preparing 'files' for '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf'
resolvers       | caching | category 'files', cachename '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf' saved in '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/28a03cdd14c63e5c76566ae30b8a381b.lua'
system          | lua | compiling '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/28a03cdd14c63e5c76566ae30b8a381b.lua' into '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/28a03cdd14c63e5c76566ae30b8a381b.luc'
system          | lua | dumping '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/28a03cdd14c63e5c76566ae30b8a381b.lua' into '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/28a03cdd14c63e5c76566ae30b8a381b.luc' stripped
resolvers       | caching | 'files' compiled to '/nix/store/n7nrzyzsyb1l7qzna5f4pl3hmj3qr603-texlive-combined-2021/share/texmf-var/luatex-cache/context/1dbe71212a391f836718d0dfc155f8d1/trees/28a03cdd14c63e5c76566ae30b8a381b.luc'
mtxrun          |
mtxrun          | elapsed lua time: 0.227 seconds
error: 1 dependencies of derivation '/nix/store/rhd20m54ijhznli6q0skb7pggh0zffw5-xyce-7.4.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/gcz6hsfqa4ij9abhmb5v5afgwy3xjkll-xyce-7.4.0.drv' failed to build
error: build of '/nix/store/gcz6hsfqa4ij9abhmb5v5afgwy3xjkll-xyce-7.4.0.drv', '/nix/store/iww79xv9ibakksq2r4iijdl1c960j73q-trilinos-12.12.1.drv', '/nix/store/rhd20m54ijhznli6q0skb7pggh0zffw5-xyce-7.4.0.drv' failed
```

fedeinthemix commented:
Hmm, interesring as it builds on my system
```
...
shrinking RPATHs of ELF executables and libraries in /nix/store/r33cky2fd5slgwbm1v1va5x8xki5p4vj-xyce-7.4.0-doc
strip is /nix/store/wzmzjrpaxfrsjc37v1km8blkxqwkdgg6-gfortran-wrapper-9.3.0/bin/strip
patching script interpreter paths in /nix/store/r33cky2fd5slgwbm1v1va5x8xki5p4vj-xyce-7.4.0-doc
checking for references to /build/ in /nix/store/r33cky2fd5slgwbm1v1va5x8xki5p4vj-xyce-7.4.0-doc...
/nix/store/l5g501109nb5xpirybcp3a7zsjnyg96a-xyce-7.4.0
```
However, this is with a checkout a few weeks old (my commits on top of `b1ac54ae9be759a5254512d3794ccf198bd81eb6`).

Is there a way to see more of the failure log?

fedeinthemix commented:
Checking the `OfBorg` logs I see errors related to timeouts
```
Certification_Tests/BUG_681_SON/bsim4_681.........passed[sh]      (Time:   6s =   2.62cs +   0vs)
building of '/nix/store/fk1pw94cdmhz0z48bjcphvk9xp9g63q7-xyce-7.4.0.drv' timed out after 3600 seconds
error: build of '/nix/store/fk1pw94cdmhz0z48bjcphvk9xp9g63q7-xyce-7.4.0.drv' failed
```
Is it possible that the error that you reported is also related to a timeout? Compilation as well as running the tests takes a significant amount of time.

talyz commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,169 @@
+, libtool_2
+, mpi
+, suitesparse
+, trilinos # if compiling with MPI so trilinos must be! Pass the correct one.
```
This should instead be an assertion. Put `withMPI` into the `passthru` attribute of `trilinos` and compare it like 

```nix
assert withMPI -> trilinos.withMPI;
```

fedeinthemix replied:
Good idea. Done. Thanks.

talyz commented:
Adding `enableParallelBuilding = true;` to both `xyce` and `trilinos` speeds up the build quite a bit on multi-core machines.

talyz commented:
Building both `xyce` and `xyce-parallel` works fine for me, at least with `enableParallelBuilding = true`.

fedeinthemix commented:
Adding `enableParallelBuilding = true` indeed speeds up compilation. Both variants `xyce` and `xyce-parallel` still build and pass all tests for me.  I've added it to my last push.

fedeinthemix commented:
I've not added `enableParallelBuilding = true` to `trilinos` as it uses `cmake` and it already builds in parallel.

talyz commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,173 @@
+, enableTests ? true
+}:
+
+assert withMPI -> (trilinos.withMPI);
```
```suggestion
assert withMPI -> trilinos.withMPI;
```

talyz replied:
```suggestion
assert withMPI -> trilinos.withMPI;
```

talyz commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,173 @@
+    find . -type f -name "*.cir" -exec chmod u+w {} \;
+    patchShebangs Netlists/ TestScripts/
+    # patch script generating functions
+    sed -i -E 's/\/usr\/bin\/env perl/${builtins.replaceStrings [ "." "/" "-" "!" ] [ "\\." "\\/" "\\-" "\\!" ] perl.outPath}\/bin\/perl/'  \
```
```suggestion
    sed -i -E 's|/usr/bin/env perl|${lib.escapeRegex perl.outPath}/bin/perl|'  \
```

talyz commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,173 @@
+    # patch script generating functions
+    sed -i -E 's/\/usr\/bin\/env perl/${builtins.replaceStrings [ "." "/" "-" "!" ] [ "\\." "\\/" "\\-" "\\!" ] perl.outPath}\/bin\/perl/'  \
+      TestScripts/XyceRegression/Testing/Netlists/RunOptions/runOptions.cir.sh
+    sed -i -E 's/\/bin\/sh/${builtins.replaceStrings [ "." "/" "-" "!" ] [ "\\." "\\/" "\\-" "\\!" ] bash.outPath}\/bin\/sh/' \
```
```suggestion
    sed -i -E 's|/bin/sh|${lib.escapeRegex bash.outPath}/bin/sh|' \
```

talyz commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,173 @@
+    TEST_ROOT="$(pwd)/../source"
+
+    # Honor the TMP variable
+    sed -i -E 's/\/tmp/\$TMP/' $TEST_ROOT/TestScripts/suggestXyceTagList.sh
```
```suggestion
    sed -i -E 's|/tmp|\$TMP|' $TEST_ROOT/TestScripts/suggestXyceTagList.sh
```

talyz commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,173 @@
+  ];
+
+  buildInputs = [
+    bison blas flex fftw lapack suitesparse trilinos
```
```suggestion
    bison
    blas
    flex
    fftw
    lapack
    suitesparse
    trilinos
```

talyz commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,173 @@
+        enumitem
+        multirow
+        preprint;
+      })
```
```suggestion
    })
```

talyz commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,173 @@
+  enableParallelBuilding = true;
+
+  nativeBuildInputs = [
+    autoconf automake gfortran libtool_2
```
```suggestion
    autoconf
    automake
    gfortran
    libtool_2
```

talyz commented on pkgs/applications/science/electronics/xyce/default.nix:
```diff
@@ -0,0 +1,173 @@
+      install -t $doc/share/doc/${pname}-${version}/ $(basename $d.pdf)
+      popd
+    done
+    '';
```
```suggestion
  '';
```

talyz commented on pkgs/development/libraries/science/math/trilinos/default.nix:
```diff
@@ -0,0 +1,96 @@
+    -DTrilinos_ENABLE_Zoltan=ON
+    -DTPL_ENABLE_MPI=ON
+  '';
+
+in stdenv.mkDerivation rec {
```
```suggestion
in
stdenv.mkDerivation rec {
```

talyz commented on pkgs/development/libraries/science/math/trilinos/default.nix:
```diff
@@ -0,0 +1,96 @@
+
+  buildInputs = [ blas boost lapack suitesparse ] ++ lib.optionals withMPI [ mpi ];
+
+  preConfigure = if withMPI then
+    ''cmakeFlagsArray+=(${flagsBase} ${flagsParallel})''
+  else
+    ''cmakeFlagsArray+=(${flagsBase})'';
```
```suggestion
  preConfigure =
    if withMPI then ''
      cmakeFlagsArray+=(${flagsBase} ${flagsParallel})
    ''
    else ''
      cmakeFlagsArray+=(${flagsBase})
    '';
```

talyz reviewed (approved):
Functionally, I think this is done. I have some suggestions to improve formatting and readability. You don't have to apply the ones you disagree with, if any.

fedeinthemix commented:
I wasn't familiar with the `sed` "s|||" syntax (not even described in the info manual) and the `escapeRegex` function. They are indeed super useful to increase readability. If there are no other suggestions I'll squash commits and push with a clean log.

fedeinthemix commented:
I cleaned up the log.

talyz commented:
> I wasn't familiar with the `sed` "s|||" syntax (not even described in the info manual) and the `escapeRegex` function. They are indeed super useful to increase readability.

Yep, they're both pretty neat. :) The `sed` syntax is actually even more general - you can use any character as a delimiter for the `s` command, so I usually pick one that isn't in the regex.

talyz commented on pkgs/development/libraries/science/math/trilinos/default.nix:
```diff
@@ -0,0 +1,99 @@
+  pname = "trilinos";
+  version = "12.12.1"; # Xyce 7.4 requires version 12.12.1
+
+  src = fetchurl {
+    url = "https://github.com/trilinos/Trilinos/archive/refs/tags/${pname}-release-${lib.replaceStrings [ "." ] [ "-" ] version}.tar.gz";
+    sha256 = "064ynimvsj17pvif8n990jr2al451mpwy9hpgr5242b3khrcax2l";
+  };
```
```suggestion
  src = fetchFromGitHub {
    owner = "trilinos";
    repo = "Trilinos";
    rev = "${pname}-release-${lib.replaceStrings [ "." ] [ "-" ] version}";
    sha256 = "sha256-Nqjr7RAlUHm6vs87a1P84Y7BIZEL0Vs/A1Z6dykfv+o=";
  };
```

talyz commented on pkgs/development/libraries/science/math/trilinos/default.nix:
```diff
@@ -0,0 +1,99 @@
+{ stdenv
+, fetchurl
+, fetchgit
+, lib
```
```suggestion
, lib
, fetchFromGitHub
```

talyz reviewed (changes requested):
Just noticed that the fetching could be done in a more standard / simpler way. When that's fixed, I think this is ready to merge.

fedeinthemix commented:
Thank you for the thorough review and merging my PR :-)

talyz commented:
No problem :)
